jobs/async-upload/job/models.py

```diff
@@ -96,18 +96,29 @@ def validate_uri_storage(self) -> 'URISourceStorageConfig':
 SourceConfig = Union[S3StorageConfig, OCIStorageConfig, URISourceStorageConfig]
 DestinationConfig = Union[S3StorageConfig, OCIStorageConfig]
 
+class UploadIntent(StrEnum):
+    create_model = "create_model"
+    create_version = "create_version"
+    update_artifact = "update_artifact"
 
 class ModelConfig(BaseModel):
     """Model registry model information."""
-    id: str = Field(..., description="Model ID")
-    version_id: str = Field(..., description="Model version ID")
-    artifact_id: str = Field(..., description="Model artifact ID")
+    upload_intent: UploadIntent = Field(UploadIntent.update_artifact, description="Model upload intent")
+    id: str | None = Field(..., description="Model ID")
+    version_id: str | None = Field(..., description="Model version ID")
+    artifact_id: str | None = Field(..., description="Model artifact ID")
 
     @model_validator(mode='after')
     def validate_model_ids(self) -> 'ModelConfig':
         """Validate that all model IDs are provided."""
-        if not all([self.id, self.version_id, self.artifact_id]):
-            raise ValueError("Model ID, version ID and artifact ID must be set")
+        if self.upload_intent == UploadIntent.create_model:
+            return self
```

Comment:
I think this is what @tarilabs was saying, but maybe we should also ensure ids are **not** set when they shouldn't be needed? Maybe it's fine to just ignore them though...

Reply:
> I think this is what @tarilabs was saying, but maybe we should also ensure ids are **not** set when they shouldn't be needed? Maybe it's fine to just ignore them though...

ignore or emit a warning
vs
ensuring are not set

either ^ work with me; my point is also to make sure the test semantic is clear if we'll be looking up into them for non regressions